src/subdomains/core/payment-link/controllers/payment-link.controller.ts

```diff
@@ -0,0 +1,95 @@
+import { Body, Controller, Delete, ForbiddenException, Get, Param, Post, Put, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth, ApiCreatedResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger';
+import { GetJwt } from 'src/shared/auth/get-jwt.decorator';
+import { JwtPayload } from 'src/shared/auth/jwt-payload.interface';
+import { RoleGuard } from 'src/shared/auth/role.guard';
+import { UserRole } from 'src/shared/auth/user-role.enum';
+import { UserDataService } from 'src/subdomains/generic/user/models/user-data/user-data.service';
+import { CreatePaymentLinkPaymentDto } from '../dto/create-payment-link-payment.dto';
+import { CreatePaymentLinkDto } from '../dto/create-payment-link.dto';
+import { PaymentLinkDtoMapper } from '../dto/payment-link-dto.mapper';
+import { PaymentLinkDto } from '../dto/payment-link.dto';
+import { UpdatePaymentLinkDto } from '../dto/update-payment-link.dto';
+import { PaymentLinkService } from '../services/payment-link.services';
+
+@ApiTags('Payment Link')
+@Controller('paymentLink')
+export class PaymentLinkController {
+  constructor(
+    private readonly userDataService: UserDataService,
+    private readonly paymentLinkService: PaymentLinkService,
+  ) {}
+
+  @Get()
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse({ type: PaymentLinkDto, isArray: true })
+  async getAllPaymentLinks(@GetJwt() jwt: JwtPayload): Promise<PaymentLinkDto[]> {
+    await this.checkPaymentLinksAllowed(+jwt.user);
+    return this.paymentLinkService.getAll(+jwt.user).then(PaymentLinkDtoMapper.toLinkDtoList);
+  }
+
+  @Get(':idOrExternalId')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  @ApiOkResponse({ type: PaymentLinkDto })
+  async getPaymentLink(
+    @GetJwt() jwt: JwtPayload,
+    @Param('idOrExternalId') idOrExternalId: string,
+  ): Promise<PaymentLinkDto> {
```

Comment:
How do we solve the problem where id and external-id are the same?

Reply:
I have changed the APIs to use query parameters "id" or "externalId"